pkgs/os-specific/linux/amneziawg/default.nix

```diff
@@ -0,0 +1,50 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  kernel,
+  nix-update-script,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "amneziawg";
+  version = "1.0.20240711";
+
+  src = fetchFromGitHub {
+    owner = "amnezia-vpn";
+    repo = "amneziawg-linux-kernel-module";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-WOcBTxetVz2Sr62c+2aGNyohG2ydi+R+az+4qHbKprI=";
+  };
+
+  sourceRoot = "${finalAttrs.src.name}/src";
+  hardeningDisable = [ "pic" ];
+  nativeBuildInputs = kernel.moduleBuildDependencies;
+
+  buildFlags = [
+    "apply-patches"
+    "module"
+  ];
+
+  makeFlags =
+    kernel.makeFlags
+    ++ lib.optional (lib.versionAtLeast kernel.version "5.6") "KERNEL_SOURCE_DIR=${kernel.src}"
+    ++ [ "KERNELDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build" ];
```

Comment:
Are those really ordering dependent? If not please move the optional to the end.

Reply:
Thanks, options are order independent. Also, can you please help with understanding nixpkgs conventions?
When it is advised to use `refs/tags/v{version}`, and when just `v{version}`? Was it worth combining all the PR about amneziawg into one?